torch/csrc/api/include/torch/nn/functional/pooling.h

```diff
@@ -849,14 +849,15 @@ inline std::tuple<Tensor, Tensor> fractional_max_pool3d_with_indices(
   c10::optional<ExpandingArray<3>> output_size_ = output_size;
   if (output_size_ == c10::nullopt) {
     TORCH_INTERNAL_ASSERT(output_ratio != c10::nullopt);
-    output_size_ = {(int64_t)(input.sizes()[2] * (*output_ratio.value())[0]),
-                    (int64_t)(input.sizes()[3] * (*output_ratio.value())[1]),
-                    (int64_t)(input.sizes()[4] * (*output_ratio.value())[2])};
+    output_size_ = {(int64_t)(input.sizes()[-3] * (*output_ratio.value())[0]),
+                    (int64_t)(input.sizes()[-2] * (*output_ratio.value())[1]),
+                    (int64_t)(input.sizes()[-1] * (*output_ratio.value())[2])};
```

Comment:
Indexing into IntArrayRef doesn't wrap around. input.sizes()[-1] actually tries to index `input.sizes()[MAX_INT - 1]` which is definitely not what you wanted

Reply:
input.size(-1) should do the trick though